lib/membrane/silence_generator.ex

```diff
@@ -0,0 +1,76 @@
+defmodule Membrane.SilenceGenerator do
+  @moduledoc """
+  Element responsible for generating silence as raw audio.
+  """
+
+  use Membrane.Source
+
+  alias Membrane.{Buffer, Time}
+  alias Membrane.Caps.Audio.Raw
+
+  def_options caps: [
+                type: :struct,
+                spec: Raw.t(),
+                description: "Audio caps"
+              ],
+              duration: [
+                type: :integer,
+                spec: Time.t(),
```

Comment:
I will add that possibility

Reply:
I think it could be just `:time` or `:timeout` (there is built-in type `timeout` meaning `:infinity | non_neg_integer()`). But it's not that important
1. This option accepts only an atom, it's not a typespec, so you cannot use pipe here
1. The `type` here was meant for parsing CLI options. Plus, it provides default `spec` and formatter for rendering default value in docs (so e.g. with type `:time` the default value of `2` would appear as `2 |> Membrane.Time.nanoseconds()` in docs). Other than that, it does not affect anything

Reply:
So i will use `:timeout` as a type and `Time.t() | atom()` as spec